v7.8.0 (#13655)
mui/mui-x: CHANGELOG.md

Reviewer: I think we might also want to release v6 too this week, preferably before releasing this one to avoid making v6 one the latest on GitHub releases page.

Author: That PR is up too https://github.com/mui/mui-x/pull/13656

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.8.0

_Jun 28, 2024_

We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:

> - 🛰 Introduce server-side data source for improved server integration in the Data Grid. See [documentation](https://mui.com/x/react-data-grid/server-side-data/) for more details.

Reviewer: Not sure if it's too descriptive, thoughts?
```suggestion
- 🛰 Introduce server-side data source for improved server integration in the Data Grid. 
  
  It supports server-side pagination, sorting, and filtering on plain and tree data along with automatic caching support.
  To enable the server-side data source, provide the `getRows` function to the `unstable_dataSource` prop on the Data Grid component which provides all the necesary information for the server to return the required data.

  ```tsx
  const dataSource = {
    getRows: async (params: GridServerGetRowsParams) => {
      const data = await fetch(
        `https://api.example.com/data?${new URLSearchParams({
          page: params.page,
          pageSize: params.pageSize,
          sortModel: JSON.stringify(params.sortModel),
          filterModel: JSON.stringify(params.filterModel),
        }).toString()}`,
      );
      return {
        rows: data.rows,
        totalRows: data.totalRows,
      };
    },

  }
  <DataGridPro
    unstable_dataSource={dataSource}
    {...otherProps}
  />
  ```

  See [plain data](https://mui.com/x/react-data-grid/server-side-data/) or [tree-data](https://mui.com/x/react-data-grid/server-side-data/tree-data/) documentation sections for more details.

```

Author: Shortened it slightly, let me know what you think https://github.com/mui/mui-x/pull/13655/commits/0192e3ea6156c89b3bb629481897309777201dec

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.8.0

_Jun 28, 2024_

We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:

- 🛰 Introduce server-side data source for improved server integration in the Data Grid. See [documentation](https://mui.com/x/react-data-grid/server-side-data/) for more details.
- 📈 Support Date data on the BarChart component
- ↕️ Support custom column sort icons on the Data Grid
- 🖱️ Support modifying the expansion trigger on the Tree View components

> - 🐞 Bugfixes
> - 📚 Documentation improvements

Reviewer 2: We usually put that when there is nothing specific to highlight

```suggestion
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.8.0

_Jun 28, 2024_

We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:

> - 📈 Support Date data on the BarChart component

Reviewer: ```suggestion
- 🛰 Introduce server-side data source for improved server integration in the Data Grid. See [documentation](https://mui.com/x/react-data-grid/server-side-data/) for more details.
- 📈 Support Date data on the BarChart component
```

> - 🖱️ Support modifying the expansion trigger on the Tree View component

Reviewer 3: Nitpick:

```suggestion
- 🖱️ Support modifying the expansion trigger on the Tree View components
```

- ↕️ Support custom column sort icons on the Data Grid

> - 📝 Add Overlays documentation for the Data Grid

Reviewer: ```suggestion
- 📝 Add "Overlays" section to the Data Grid documentation
```

- 🐞 Bugfixes
- 📚 Documentation improvements

<!--/ HIGHLIGHT_ABOVE_SEPARATOR /-->

### Data Grid

#### `@mui/x-data-grid@7.8.0`

- [DataGrid] Add `columnHeaderSortIcon` slot (#13563) @arminmeh
- [DataGrid] Fix dimensions lag issue after autosize (#13587) @MBilalShafi
- [DataGrid] Fix print export failure when `hideFooter` option is set (#13034) @tarunrajput

> - [DataGrid][docs] Add overlay documentation (#13624) @KenanYusuf

Reviewer 3: Should probably move to the doc and be changed to:

```suggestion
- [docs] Add documentation for the Data Grid's overlays (#13624) @KenanYusuf
```

#### `@mui/x-data-grid-pro@7.8.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@7.8.0`, plus:

> - [DataGridPro] Fix multi-sorting indicator being cut off (#13625) @KenanYusuf

Reviewer: ```suggestion
- [DataGridPro] Fix multi-sorting indicator being cut off (#13625) @KenanYusuf
- [DataGridPro] Server-side tree data support (#12317) @MBilalShafi
```

#### `@mui/x-data-grid-premium@7.8.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@7.8.0`.

### Date and Time Pickers

> #### `@mui/x-date-pickers@7.8.0`

Reviewer 3: We can remove the whole picker section 
We have to release the packages because of the license version bump though

#### `@mui/x-date-pickers-pro@7.8.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@7.8.0`.

### Charts

#### `@mui/x-charts@7.8.0`

- [charts] Fix line chart props not passing correct event handlers (#13609) @JCQuintas
- [charts] Support BarChart with `Date` data (#13471) @alexfauquette

> - [charts] Support RTL for y-axis (#13614) @alexfauquette

Reviewer 4: ```suggestion
- [charts] Support RTL for y-axis (#13614) @alexfauquette
- [charts] Use default values instead of non-null assertion to prevent error being thrown (#13637) @JCQuintas
```

### Tree View

#### `@mui/x-tree-view@7.8.0`

- [TreeView] Add `expansionTrigger` prop (#13533) @noraleonte
- [TreeView] Support experimental features from plugin's dependencies (#13632) @flaviendelangle

### Docs

- [docs] Add callout for `Luxon` `throwOnInvalid` support (#13621) @LukasTy

### Core

- [core] Add eslint rule to restrict import from `../internals` root (#13633) @JCQuintas

> - [chart] Use default values instead of non-null assertion to prevent error being thrown (#13637) @JCQuintas

Reviewer 4: ```suggestion
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.8.0

_Jun 28, 2024_

We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:

- 🛰 Introduce server-side data source for improved server integration in the Data Grid. See [documentation](https://mui.com/x/react-data-grid/server-side-data/) for more details.
- 📈 Support Date data on the BarChart component
- 🖱️ Support modifying the expansion trigger on the Tree View components
- ↕️ Support custom column sort icons on the Data Grid

> - 📝 Add "Overlays" section to the Data Grid documentation

Reviewer 3: It's up for debate, but I think we have too many highlights and not all of them are impact-full enough to be here.

Do we usually add highlights for new doc sections?
Same for the `Date` data support on BarChart, it makes it look like a major new feature but the PR is super small (#13471)
I could even drop the custom expansion trigger highlight, they are all super small additions compared with the data source.

Author: I'm not too sure what constitutes as impactful enough yet, so happy to take the consensus here.

Reviewer 3: I'm not 100% confident saying which features of other teams is impactful enough.

For eXplore, I'd say nothing this week, on a small release I would keep the "expansion trigger" but it's more to have something to say.

Reviewer: I'd agree that not adding super-smaller changes to the highlights would put more emphasis on the comparatively larger feature with a much higher anticipated audience. 👍 
But I am not sure which ones to keep and which ones to remove.

We could also split the data source highlight into multiple lines and add a bit more detail to improve the emphasis. 🤔

Reviewer 3: Totally agree that the data source is worth a code snippet :ok_hand: (usually we use a GIF but it's not really a design feature)

Reviewer 2: > Same for the Date data support on BarChart, it makes it look like a major new feature but the PR is super small

I don't think the PR size matters. I added it in the changelog because it impacts how charts are rendered if you use `Date` object. I theory everybody should be happy about this modification, but I prefer to have a highlight to let user see their is a modification

Reviewer 3: OK if you think the `Date` support is worth highlighting, you clearly know better than me :+1:

        totalRows: data.totalRows,
      };
    },
  }
  <DataGridPro
    unstable_dataSource={dataSource}
    {...otherProps}
  />
  ```

  See [server-side data documentation](https://mui.com/x/react-data-grid/server-side-data/) for more details.

- 📈 Support Date data on the BarChart component
- ↕️ Support custom column sort icons on the Data Grid
- 🖱️ Support modifying the expansion trigger on the Tree View components

<!--/ HIGHLIGHT_ABOVE_SEPARATOR /-->

### Data Grid

#### `@mui/x-data-grid@7.8.0`

- [DataGrid] Add `columnHeaderSortIcon` slot (#13563) @arminmeh
- [DataGrid] Fix dimensions lag issue after autosize (#13587) @MBilalShafi
- [DataGrid] Fix print export failure when `hideFooter` option is set (#13034) @tarunrajput

#### `@mui/x-data-grid-pro@7.8.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@7.8.0`, plus:

- [DataGridPro] Fix multi-sorting indicator being cut off (#13625) @KenanYusuf
- [DataGridPro] Server-side tree data support (#12317) @MBilalShafi

#### `@mui/x-data-grid-premium@7.8.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

> Same changes as in `@mui/x-data-grid-pro@7.8.0`.

Reviewer 5: Sorry for the late intrusion, but I've merged #13652. 🙈 

```suggestion
Same changes as in `@mui/x-data-grid-pro@7.8.0`.

### Date and Time Pickers

#### `@mui/x-date-pickers@7.8.0`

- [fields] Fix section clearing behavior on Android (#13652) @LukasTy

#### `@mui/x-date-pickers-pro@7.8.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@7.8.0`.
```

### Charts

#### `@mui/x-charts@7.8.0`

- [charts] Fix line chart props not passing correct event handlers (#13609) @JCQuintas
- [charts] Support BarChart with `Date` data (#13471) @alexfauquette
- [charts] Support RTL for y-axis (#13614) @alexfauquette
- [charts] Use default values instead of non-null assertion to prevent error being thrown (#13637) @JCQuintas

### Tree View

#### `@mui/x-tree-view@7.8.0`

- [TreeView] Add `expansionTrigger` prop (#13533) @noraleonte
- [TreeView] Support experimental features from plugin's dependencies (#13632) @flaviendelangle

### Docs

- [docs] Add callout for `Luxon` `throwOnInvalid` support (#13621) @LukasTy
- [docs] Add "Overlays" section to the Data Grid documentation (#13624) @KenanYusuf

### Core

- [core] Add eslint rule to restrict import from `../internals` root (#13633) @JCQuintas

> - [docs-infra] Sync \_app file with monorepo (#13582) @Janpot

Reviewer 5: ```suggestion
- [docs-infra] Sync `\_app` folder with monorepo (#13582) @Janpot
```

> - [license] Allow usage of charts an tree view pro package for old premium licenses (#13619) @flaviendelangle

Reviewer 5: ```suggestion
- [license] Allow usage of charts and tree view pro package for old premium licenses (#13619) @flaviendelangle
```

- 📄 Support [unknown and estimated row count in server-side pagination](https://mui.com/x/react-data-grid/pagination/#index-based-pagination) (#12490) @MBilalShafi
- 🎨 Support color scales in Charts (#12490) @alexfauquette
  Add a [`colorMap` configuration](https://mui.com/x/react-charts/styling/#values-color) to an axis, and the chart will use it to select colors.

>   Each impacted chart ([bar charts](https://mui.com/x/react-charts/bars/#color-scale), [line charts](https://mui.com/x/react-charts/lines/#color-scale), [scatter chasrts](https://mui.com/x/react-charts/scatter/#color-scale)) has a dedicated section explaining how this color map is impacting it.

Reviewer 5: ```suggestion
  Each impacted chart ([bar charts](https://mui.com/x/react-charts/bars/#color-scale), [line charts](https://mui.com/x/react-charts/lines/#color-scale), [scatter charts](https://mui.com/x/react-charts/scatter/#color-scale)) has a dedicated section explaining how this color map is impacting it.
```
